frame/merkle-mountain-range/src/lib.rs

```diff
@@ -196,13 +205,32 @@ pub mod pallet {
 	#[pallet::getter(fn mmr_leaves)]
 	pub type NumberOfLeaves<T, I = ()> = StorageValue<_, LeafIndex, ValueQuery>;
 
-	/// Hashes of the nodes in the MMR.
+	/// Leaf data added by current block (leaf index: `NumberOfLeaves-1`).
+	///
+	/// This is single value storage overwritten by each block. It is used as temporary
+	/// leaf storage until after block is built and offchain worker can move/copy this
+	/// leaf in offchain database under `node_offchain_key()`.
+	#[pallet::storage]
+	#[pallet::getter(fn latest_leaf)]
+	pub type LatestLeaf<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, BoundedVec<u8, T::MaxLeafSize>, ValueQuery>;
+
+	/// TODO
+	#[pallet::storage]
+	#[pallet::getter(fn new_nodes)]
+	pub type NewNodes<T: Config<I>, I: 'static = ()> = StorageValue<
+		_,
+		BoundedVec<(NodeIndex, <T as Config<I>>::Hash), T::MaxMmrHeight>,
+		ValueQuery,
+	>;
```

Comment:
Would love to unify these two in a single:

```suggestion
	#[pallet::storage]
	#[pallet::getter(fn new_nodes)]
	pub type NewNodes<T: Config<I>, I: 'static = ()> = StorageValue<
		_,
		BoundedVec<(NodeIndex, NodeOf<T, I, LeafOf<T,I>>), T::MaxMmrHeight>,
		ValueQuery,
	>;
```

but I can't seem to implement `parity_scale_codec::MaxEncodedLen` for [DataOrHash](https://github.com/paritytech/substrate/blob/df81976c40e2e0573d59c51e12eb3c15c3ff3057/primitives/merkle-mountain-range/src/lib.rs#L181) primitive.
The `DataOrHash` encoding implementations are a big spaghetti that I can't make sense of 😢

Reply:
I unified using yet another custom wrapper to circumvent needing `MaxEncodedLen` on `DataOrHash`.